src/biotite/sequence/profile.py

```diff
@@ -280,3 +291,88 @@ def _general_to_consensus(self):
         consensus = GeneralSequence(self.alphabet)
         consensus.code = np.argmax(self.symbols, axis=1)
         return consensus
+
+    def probability_matrix(self, pseudocount=0):
+        """
+        Calculate the position probability matrix (ppm) based on 'symbols'
+        and the given pseudocount.
+        This new matrix has the same shape as 'symbols'.
+
+        Parameters
+        ----------
+        pseudocount: int
+            Amount added to the number of observed cases in order to change
+            the expected probability.
+            (Default: 0)
+
+        Returns
+        -------
+        ppm: ndarray, dtype=float, shape=(n,k)
+            Calculated the position probability matrix.
+        """
+        if pseudocount < 0:
+            raise ValueError(
+                f"Pseudocount can not be smaller than zero."
+            )
+        return (self.symbols + pseudocount / len(self.symbols)) / \
+               (np.sum(self.symbols, axis=1)[:, np.newaxis] + pseudocount)
+
+    def log_odds_matrix(self, background_frequencies=None, pseudocount=0):
+        """
+        Calculate the position weight matrix (pwm) based on the
+        position probability matrix (ppm) (with given pseudocount) and
+        background_frequencies.
+        This new matrix has the same shape as 'symbols'.
+
+        Parameters
+        ----------
+        pseudocount: int
+            Amount added to the number of observed cases in order to change
+            the expected probability.
+            (Default: 0)
+        background_frequencies: float
+            (Default: 1/len(alphabet))
+        Returns
+        -------
+        pwm: ndarray, dtype=float, shape=(n,k)
+            Calculated the position weight matrix.
+        """
+        if background_frequencies is None:
+            background_frequencies = 1 / len(self.alphabet)
+        ppm = self.probability_matrix(pseudocount)
+        return np.log2(ppm / background_frequencies)
+
+    def sequence_probability_from_matrix(self, sequence, matrix):
+        """
+        Calculate probability of a sequence based on either the
+        position probability matrix (ppm) or the position weight matrix (pwm).
```

Comment:
Could you reference the mathematical background for the calculation the sequence probability from a position weight matrix? In my intuition the product of values from the position weight matrix does not represent a probability, since each value represents the logarithm of a probability, but I may be wrong.